src/ngraph/pass/constant_folding.cpp

```diff
@@ -763,3 +766,176 @@ void pass::ConstantFolding::construct_constant_quantize()
     this->add_matcher(
         quantize_matcher, constant_quantize_callback, PassProperty::REQUIRE_STATIC_SHAPE);
 }
+
+template <typename TI, typename TO>
+shared_ptr<op::Constant> fold_constant_convert_helper1(shared_ptr<op::Constant> constant,
+                                                       const element::Type& output_element_type)
+{
+    auto out_shape = constant->get_shape();
+    vector<TO> out_vec(shape_size(out_shape));
+
+    runtime::reference::convert<TI, TO>(
+        constant->get_vector<TI>().data(), out_vec.data(), shape_size(out_shape));
+
+    return make_shared<op::Constant>(output_element_type, out_shape, out_vec);
+}
+
+template <typename TI>
+shared_ptr<op::Constant> fold_constant_convert_helper0(shared_ptr<op::Constant> constant,
+                                                       const element::Type& output_element_type)
+{
+#pragma GCC diagnostic push
```

Comment:
GCC 4.8.X has some missed with handling some pragmas. Workaround shown in https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/constant.hpp#L262-L266 is needed.